core/src/com/unciv/logic/civilization/PolicyManager.kt

```diff
@@ -96,15 +96,15 @@ class PolicyManager {
 
     private fun getRulesetPolicies() = civInfo.gameInfo.ruleSet.policies
 
-    fun getPolicyByName(name: String): Policy = getRulesetPolicies()[name]!!
+    private fun getPolicyByName(name: String): Policy = getRulesetPolicies()[name]!!
```

Comment:
... and since Jack separated cleanly by commit: easy peasy to switch approaches. I'm in favour of silencing the PropertyMethodOrWhateverCouldBePrivate inspections, but which way... We could even suppress project-wide if we wished to (Not really, I know). PolicyManager is probably fine to privaticize, but overall it's dependent on context and the answer won't be the same everywhere.

Reply:
> methods may not be used outside of their classes, but they still make sense to have public. 

Actually, I do not see any sense of having a method "public" until it is used outside. If you are going to use it from outside and it has been designed for that - convert to "public" but until that I would prefer to keep it "private".
There are a lot of articles in the Internet why it is important: [here](https://www.learnhowtoprogram.com/c-and-net/objects-and-methods/encapsulation-and-visibility) and [here](http://www.cems.uwe.ac.uk/~jsa/UMLJavaShortCourse09/CGOutput/Unit3/unit3(0809)/page_13.htm) and [here](https://medium.com/lets-do-it-pl/what-is-encapsulation-and-what-are-the-benefits-of-it-e7548debd7be) and many others.

Reply:
> \@SomeTroglodyte, who added the `getPolicyByName` function

[See e7639c9](https://github.com/yairm210/Unciv/commit/e7639c93c4dc94d222aed3b50e954202c0cee58d)

Reply:
> There's a pretty large chance that for some reason, some code might decide in the future to need.....

That's a perfect illustration for the violation of the [YAGNI ](https://martinfowler.com/bliki/Yagni.html) principle. [Ron Jeffries](https://en.wikipedia.org/wiki/Ron_Jeffries) also would not agree with you: _"Always implement things when you actually need them, never when you just foresee that you need them."_

Spending your time for writing the code, which you will not use right now, increases the amount of code you need to maintain in the future - that's a shoot in an own leg.

Reply:
> YAGNI doesn't apply because the code is already there and is staying there.

It was violated when "decided on purpose to make the method public" creating it, with a mind "to be used at some point somewhere outside later".

If both you and @SomeTroglodyte strongly believe that this method must be public, of course, I will revert it to public and add a suppression there - but that goes against of my understanding of the good code.

Reply:
No I'm neutral. Or - the original implementor is the ultimate authority (which I'm not for this getPolicyByName). Some of your `private` are obviously good, some are 'well if it hasn't been used elsewhere for months...' and some I do doubt but the negative consequences are not that dangerous to playing remaining fun, so I'd go along. Or - I respect strong convictions on academic perfection by knowledgeable coders, but if they make it a bloody conflict when the product will run fine regardless then I'm :switzerland: ... Sorry for the teasing :zany_face: :upside_down_face: